src/relay/pushservice/pushservice.py

```diff
@@ -116,41 +120,45 @@ def check_client_token(self, client_token: str) -> bool:
         return True
 
 
-def _build_event_message(
-    client_token: str, event: Event
-) -> Optional[messaging.Message]:
+def _build_data_message(client_token: str, event: Event) -> Optional[messaging.Message]:
+    data = _build_data_prop(event=event)
 
-    notification = _build_notification(event)
+    if data is not None:
+        android_config = messaging.AndroidConfig(priority="high")
+        apns = messaging.APNSConfig(
+            payload=messaging.APNSPayload(aps=messaging.Aps(content_available=True),),
+        )
 
-    if notification is not None:
         return messaging.Message(
-            notification=_build_notification(event), data={}, token=client_token
+            data=data, token=client_token, android=android_config, apns=apns
         )
-    else:
-        return None
+
+    return None
 
 
-def _build_notification(event: Event) -> messaging.Notification:
-    notification = None
-    if isinstance(event, TransferEvent):
+def _build_data_prop(event: Event):
+    data = None
+
+    if isinstance(event, TrustlineUpdateEvent):
+        data = _get_data_prop_dict(event)
+    elif isinstance(event, TransferEvent):
         if event.direction == "received":
-            notification = messaging.Notification(
-                title="Payment received", body="Click for more details"
-            )
+            data = _get_data_prop_dict(event)
     elif isinstance(event, TrustlineRequestEvent):
         if event.direction == "received":
-            notification = messaging.Notification(
-                title="Trustline Update Request",
-                body="Someone wants to update a trustline",
-            )
-    elif isinstance(event, TrustlineUpdateEvent):
-        notification = messaging.Notification(
-            title="Trustline Update", body="A trustline was updated"
-        )
+            data = _get_data_prop_dict(event)
     elif isinstance(event, MessageEvent):
         if event.type == "PaymentRequest":
-            notification = messaging.Notification(
-                title="Payment Request", body="Click for more details"
-            )
+            data = {"message": event.message, "eventType": event.type}
+
+    return data
+
 
-    return notification
+def _get_data_prop_dict(event):
+    return {
+        "blockHash": str(event.block_hash.hex()),
```

Comment:
You may want to prepend `0x`, I don't think it will be there otherwise.

Reply:
the 0x should be there - otherwise this here https://github.com/trustlines-protocol/relay/pull/526/files#diff-d98cf108481d4c897e05c4cf013b9279R131 shouldn't have passed?